docs/administration/managing-infrastructure/rotate-master-key.md

```diff
@@ -0,0 +1,77 @@
+---
+title: Rotating the Master Key
+description: A guide to rotating the Master Key on all machines hosting the Octopus Server.
+position: 45
+---
+
+There are times you might want to rotate your master key, for example if you're worried about the existing master key being leaked. This guide walks you through this process. The rotation should have no impact once it's completed.
+
+This guide assumes you still have access to your Master Key. You should be able to run the [`show-master-key` command](/docs/octopus-rest-api/octopus.server.exe-command-line/show-master-key.md). If you've lost access to the Master Key, please refer to [Recovering after losing your Octopus Server and Master Key](/docs/administration/managing-infrastructure/lost-master-key.md).
+
+## What is affected by the rotation
+
+Octopus [encrypts important and sensitive data](/docs/security/data-encryption.md) using a Master Key. This includes:
+
+- The Octopus Server X.509 certificate which is used for [Octopus to Tentacle communication](/docs/security/octopus-tentacle-communication/index.md) - your Tentacles should still trust your Octopus Server after the rotation.
```

Comment:
@veochen-octopus  "Should" does not give me greater confidence. Have we tested all this and can assure users it **does** work?

Reply:
Also what happens if the tentacles are restarted?

Reply:
So I understand, the master key is used to create the certificate and once it is created, it is ok to change the master key.
This would only be a problem if we need to rotate the certificate, correct?

Reply:
>So I understand, the master key is used to create the certificate and once it is created, it is ok to change the master key.
This would only be a problem if we need to rotate the certificate, correct?

Correct.

> Also what happens if the tentacles are restarted?

Restarting tentacles should make no difference, since the cert and the thumbprint are the same after the rotation. As far as tentacles are concerned, nothing has happened to the server.

> "Should" does not give me greater confidence. Have we tested all this and can assure users it does work?

Yes this has been tested. I'm gonna leave "should" in there for now...as we can't guarantee 100%. But keen to hear what others think @zentron @harrisonmeister

Reply:
I'm fine with the wording, but will defer to @zentron from a technical perspective on likely outcomes of it not working.

Reply:
I'm happy to change it to `will` if it provides more confidence. Based on how it works it certainly _should_, so if it doesnt then theres a bug that needs reporting.